Library/Homebrew/dev-cmd/livecheck.rb

```diff
@@ -51,7 +51,7 @@ def livecheck
     formulae_to_check = if args.tap
       Tap.fetch(args.tap).formula_names.map { |name| Formula[name] }
     elsif args.installed?
-      Formula.installed
+      Formula.installed.map { |formula| Formula[formula.full_name] }
```

Comment:
At least to me, it seems like the main issue is that our assumptions around `#head?`aren't correct all the time. `#head?`, `#stable?`, etc. are only telling us what the `active_spec` is, so `head?` isn't a reliable way of identifying _only_ head-only formulae.

One way of resolving this (instead of modifying `Formula#installed`) is to add a `#head_only?` method, like we discussed previously. Something like:

```ruby
def head_only?
  head && !stable
end
```

When I added this to `Formula` and replaced instances of `formula.head?` in livecheck with `formula.head_only?`, this situation is resolved. That is to say:

* A formula with `stable` and `head` that's installed as `head` will do a normal check instead of the head-only check.
* An installed head-only formula will do the head-only check, as expected.

Reply:
Thanks Sam, this was indeed the case. I'll update the PR to include the `head_only?` method.